src/ui_imgui.hh

```diff
@@ -2765,6 +2768,295 @@ namespace UiAnimal {
+                    if (equipped) {
+                        label += " (Equipped)";
+                    }
+            
+                    if (ImGui::Selectable(label.c_str())) {
+                        if (equipped) {
+                            equipment.erase(std::remove(equipment.begin(), equipment.end(), x), equipment.end());
+                        } else {
+                            equipment.push_back(x);
+                        }
+                    }
+                }
+                ImGui::EndListBox();
+            }
+            
+    
+            // Buttons
+            ImGui::Separator();
+            if (ImGui::Button("Apply")) {
+                std::vector<int> equippedItems;
+            
+                auto push_if_valid = [&](uint8_t itemId) {
+                    if (itemId != 255 && itemId >= 0) {
+                        equippedItems.push_back(itemId);
+                    }
+                };
+            
+                push_if_valid(e->properties[ROBOT_PROPERTY_HEAD].v.i8);
+                push_if_valid(e->properties[ROBOT_PROPERTY_HEAD_EQUIPMENT].v.i8);
+                push_if_valid(e->properties[ROBOT_PROPERTY_BACK].v.i8);
+                push_if_valid(e->properties[ROBOT_PROPERTY_FRONT].v.i8);
+                push_if_valid(e->properties[ROBOT_PROPERTY_FEET].v.i8);
+                push_if_valid(e->properties[ROBOT_PROPERTY_BOLT_SET].v.i8);
+                for (uint32_t itemId : equipment) {
+                    equippedItems.push_back(itemId);
+                }
+            
+                std::stringstream ss;
+                for (size_t i = 0; i < equippedItems.size(); ++i) {
+                    if (i > 0) ss << ";";
+                    ss << equippedItems[i];
+                }
+                e->set_property(ROBOT_PROPERTY_EQUIPMENT, ss.str().c_str());
+                ui::message("Robot properties saved!");
+                P.add_action(ACTION_HIGHLIGHT_SELECTED, 0);
+                P.add_action(ACTION_RESELECT, 0);
+                W->add_action(e->id, ACTION_CALL_ON_LOAD);
+                ImGui::CloseCurrentPopup();
+                last_selected = nullptr;
+                needs_refresh = true;
+            }
+            ImGui::SameLine();
+            if (ImGui::Button("Cancel")) {
+                ImGui::CloseCurrentPopup();
+                last_selected = nullptr;
+                needs_refresh = true;
+            }
+    
+            ImGui::EndPopup();
+        }
+    }
+}
+
+namespace UiSticky {
+    static bool do_open = false;
+
+    static void open() {
+        do_open = true;
+    }
+
+    static void layout() {
+        entity* e = G->selection.e;
+
+        static bool center_x = true;
+        static bool center_y = true;
+        static int font_size = 2;
+        static std::string text = "Hello!";
+
+        if (do_open) {
+            text = e->properties[0].v.s.buf ? e->properties[0].v.s.buf : "Hello!";
```

Comment:
Put initialisation stuff in open() instead of in layout behind an if check, like is done in other dialogs